src/Lachain.Storage/Repositories/LocalTransactionRepository.cs

```diff
@@ -33,9 +33,22 @@ public void TryAddTransaction(TransactionReceipt receipt)
         {
             if (_watchAddresses.Count(addr =>
                     addr.Equals(receipt.Transaction.To) || addr.Equals(receipt.Transaction.From)) <= 0) return;
-            
-            var data = LoadState();
-            SaveState(data.Concat(receipt.Hash.ToBytes()).ToArray());
+
+            if (receipt.Transaction.Invocation.Length > 0)
```

Comment:
We will return at the previous line if tx contains our address in Invocation only,  so this condition will not be checked in this case